Source/NSStoryboard.m

```diff
@@ -0,0 +1,237 @@
+                    }
+                  
+                  // Remove the appNode
+                  [appNode detach];
+                  
+                  /*
+                    <customObject id="-2" userLabel="File's Owner" customClass="NSApplication">
+                      <connections>
+                        <outlet property="delegate" destination="Voe-Tx-rLC" id="GzC-gU-4Uq"/>
+                      </connections>
+                    </customObject>
+                  */
+                  
+                  // create a customObject entry for NSApplication reference...
+                  NSXMLElement *customObject = [[NSXMLElement alloc] initWithName: @"customObject"];
+                  NSXMLNode *idValue   = [NSXMLNode attributeWithName: @"id"
+                                                          stringValue: @"-2"];
+                  NSXMLNode *usrLabel  = [NSXMLNode attributeWithName: @"userLabel"
+                                                          stringValue: @"File's Owner"];
+                  NSXMLNode *customCls = [NSXMLNode attributeWithName: @"customClass"
+                                                          stringValue: @"NSApplication"];
+                  [customObject addAttribute: idValue];
+                  [customObject addAttribute: usrLabel];
+                  [customObject addAttribute: customCls];
+                  
+                  // Add it to the document
+                  [objects addChild: customObject];
+                  [objects detach];
+                  [doc addChild: objects];
+                  RELEASE(customObject);
+                  
+                  // Assign application scene...
+                  ASSIGN(_applicationSceneId, sceneId);
+                }
+              else
+                {
+                  [child detach];
+                  [doc addChild: child];
+                }
+              
+              // Create document...
+              document = [[NSXMLDocument alloc] initWithRootElement: doc];
+              [_scenesMap setObject: document
+                             forKey: sceneId];
+              RELEASE(document);
+            }
+        }
+    }
+  else
+    {
+      NSLog(@"No document element in storyboard file");
+    }
+
+  NSLog(@"map = %@", _scenesMap);
+  NSLog(@"initial = %@", _initialViewControllerId);
+}
+
+// Private instance methods...
+- (id) initWithName: (NSStoryboardName)name
+             bundle: (NSBundle *)bundle
+{
+  self = [super init];
+  if (self != nil)
+    {
+      NSString *path = [bundle pathForResource: name
+                                        ofType: @"storyboard"];
+      NSData *data = [NSData dataWithContentsOfFile: path];
+      
+      NSXMLDocument *storyboardXml = [[NSXMLDocument alloc] initWithData: data
+                                                                 options: 0
+                                                                   error: NULL];
+      AUTORELEASE(storyboardXml);
```

Comment:
Why not move this after the next line and make it a RELEASE?

Reply:
DONE